src/index.ts

```diff
@@ -2,27 +2,33 @@ import * as engine from './engine';
 import { getPageFeatures } from './features';
 import { viewStore, matchedAudienceStore } from './store';
 import { timeStampInSecs } from './utils';
-import { hasGdprConsent } from './gdpr';
+import { waitOnConsent } from './gdpr';
 import {
   PageFeatureGetter,
   MatchedAudience,
   AudienceDefinition,
 } from '../types';
 
+export { runOnConsent, waitOnConsent } from './gdpr';
+
 interface Config {
   pageFeatureGetters: PageFeatureGetter[];
   audienceDefinitions: AudienceDefinition[];
   vendorIds?: number[];
   omitGdprConsent?: boolean;
+  allowMultipleRuns?: boolean; // For testing purposes only
 }
 
+let hasRun = false;
 // TODO: we need to give a way to consumers to ensure this does not
 // run multiple times on a single page load.
 const run = async (config: Config): Promise<void> => {
-  if (config.omitGdprConsent !== true) {
-    const hasConsent = await hasGdprConsent(config.vendorIds);
-    if (!hasConsent) return;
+  if (hasRun && config.allowMultipleRuns !== true) {
+    return Promise.resolve();
   }
+  hasRun = true;
+
+  await waitOnConsent(config.vendorIds, config.omitGdprConsent);
```

Comment:
I assume the result of this should be captured?

Reply:
It's not a boolean, it'll wait until there is consent. For example: if there's no consent granted, and maybe the user is on the page for 15 mintues and then decides to grant consent, it will continue off from this point because now consent is granted.

I'm not sure about the design of this though, because the run function will "hang" until consent is granted. Maybe that's okay?

Reply:
Hmm ok got you. Seems to address the needed behaviour.

Is there a max time something can be `awaited` ?

Reply:
Nope, I think it'll just keep going until it's resolved